RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -415,6 +422,26 @@ namespace {
     }
   }  // namespace dnn_inputs_2017_v2
 
+  float getTauID(const pat::Tau& tau, const std::string& tauID, float default_value = -999.)
+  {
+    static std::set<std::string> isFirstWarning;
```

Comment:
Ah no, I missed that it is static, sry

Reply:
Yes, it works, because the isFirstWarning variable is static.

Reply:
@veelken sorry, I've missed it during the review: operations on `std::set` are not thread-safe. If we want to have a static set, a mutex to access it should also be implemented.

Reply:
@kandrosov  I have added the mutex. Unfortunately, I cannot commit the changes via GitHub anymore (because I get many merge conflicts between my private CMSSW repository and the Tau POG one when I pull)

Can one of you please take the changes from /afs/cern.ch/user/v/veelken/public/DeepTauId.cc ?
There are only 3 lines of code that have changed [*].

- - - - - - - -
[*]

git diff plugins/DeepTauId.cc
diff --git a/RecoTauTag/RecoTau/plugins/DeepTauId.cc b/RecoTauTag/RecoTau/plugins/DeepTauId.cc
index a0e8ece7ef1..77ceeff87b6 100644
--- a/RecoTauTag/RecoTau/plugins/DeepTauId.cc
+++ b/RecoTauTag/RecoTau/plugins/DeepTauId.cc
@@ -14,6 +14,9 @@
 #include <TString.h>  // Form
 
 #include <fstream>
+#include <mutex>
+
+std::mutex g_mutex;
 
 namespace deep_tau {
   constexpr int NumberOfOutputs = 4;
@@ -421,6 +424,7 @@ namespace {
   }  // namespace dnn_inputs_2017_v2
 
   float getTauID(const pat::Tau& tau, const std::string& tauID, float default_value = -999.) {
+    std::lock_guard<std::mutex> guard(g_mutex);
     static std::set<std::string> isFirstWarning;
     if (tau.isTauIDAvailable(tauID)) {
       return tau.tauID(tauID);

Reply:
Sure, I will take care of adding this.